### DISCUSSION TITLE
Received an empty query

### DISCUSSION THREAD

**Discussion by rwxroot** [diesel-rs/diesel#3205] [Q&A] [answered]
We have been using diesel in production at my workplace, and recently, we integrated the [http://docs.diesel.rs/diesel_migrations/fn.revert_latest_migration_in_directory.html](revert_latest_migration_in_directory) function into our code base. Upon not providing any queries in down.sql, we have been getting `Err(QueryError(DatabaseError(__Unknown, "Received an empty query")))` error. 

Is it essential to provide queries in down.sql? Is there a way to bypass this without having to put dummy queries?

**Comment by weiznich** [comment, marked as answer]
This is expected behaviour. If you revert an migration it's always expected to have an down migration, as otherwise reapplying the migration again wouldn't be possible.

**Reply by rwxroot** [reply]
Yes that makes sense but what if there is absolutely nothing that I can put in the down.sql? Is there a way I can get away with it?

**Reply by weiznich** [reply]
I would like to see an example for this. Any migration has likely side effects. This means a `down` migration is required, as otherwise `diesel migration redo` will not work.

**Reply by rwxroot** [reply]
Initially, we used to manipulate our database with hasura as well, which was later stopped and now we only do that through diesel. In that period, some tables were created, which were later deemed unnecessary and were dropped in up.sql, but it doesn't make sense to add them back in down.sql because we don't want them. To overcome this problem, we added a dummy query, so we don't break the down migration flow. Also, we have not integrated `redo` into our system, so that's not required, at least for now.

**Reply by weiznich** [reply]
I continue to feel that this is an antipattern, but I might be fine with a PR that changes the behaviour for empty migration so that they do not create an error, but emit a warning instead. At least as long as the PR contains tests + documentation for this behaviour.

**Reply by rwxroot** [reply]
Alright @weiznich, I'll get on it ASAP and thank you for your time and support.
